src/unp4k/Program.cs

```diff
@@ -1,100 +1,79 @@
-						{
-							var target = new FileInfo(entry.Name);
-
-							if (!target.Directory.Exists) target.Directory.Create();
-
-							if (!target.Exists)
-							{
-								Console.WriteLine($"{entry.CompressionMethod} | {crypto} | {entry.Name}");
-
-								using (Stream s = pak.GetInputStream(entry))
-								{
-									using (FileStream fs = File.Create(entry.Name))
-									{
-										StreamUtils.Copy(s, fs, buf);
-									}
-								}
-
-								// target.Delete();
-							}
-						}
-					}
-					catch (Exception ex)
-					{
-						Console.WriteLine($"Exception while extracting {entry.Name}: {ex.Message}");
+            using FileStream pakFile = File.OpenRead(args[0]);
+            ZipFile pak = new ZipFile(pakFile);
+            byte[] buf = new byte[4096];
 
-						try
-						{
-							using (var client = new HttpClient { })
-							{
-								// var server = "http://herald.holoxplor.local";
-								var server = "https://herald.holoxplor.space";
+            foreach (ZipEntry entry in pak)
+            {
+                try
+                {
+                    if (args[1].StartsWith("*.")) args[1] = args[1].Substring(1);                                                                                           // Enable *.ext format for extensions
+                    if (args[1] == ".*" ||                                                                                                                                  // Searching for everything
+                        args[1] == "*" ||                                                                                                                                   // Searching for everything
+                        entry.Name.ToLowerInvariant().Contains(args[1].ToLowerInvariant()) ||                                                                               // Searching for keywords / extensions
+                        (args[1].EndsWith("xml", StringComparison.InvariantCultureIgnoreCase) && entry.Name.EndsWith(".dcb", StringComparison.InvariantCultureIgnoreCase))) // Searching for XMLs - include game.dcb
+                    {
+                        FileInfo target = new FileInfo(entry.Name);
+                        if (!target.Directory.Exists) target.Directory.Create();
+                        if (!target.Exists)
+                        {
+                            Console.WriteLine($"{entry.CompressionMethod} | {entry.Name}");
+                            using Stream s = pak.GetInputStream(entry);
+                            using FileStream fs = File.Create(entry.Name);
+                            StreamUtils.Copy(s, fs, buf);
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine($"Exception while extracting {entry.Name}: {ex.Message}");
+                    /*
```

Comment:
Was old debug reporting - has been decommissioned

Reply:
Ah right, ill remove it, thanks.